### PR TITLE
Add other client extensions

### LINKED ISSUE #613: INatsClient KV API extensions
### Proposed change

INatsClient KV API extensions

### Use case

INatsClient KV API extensions

Note: Expose INatsJSContext as a property

### LINKED ISSUE #614: INatsClient Object Store API extensions
### Proposed change

INatsClient Object Store API extensions

### Use case

INatsClient Object Store API extensions

Note: Expose INatsJSContext as a property

### LINKED ISSUE #615: INatsClient Services API extensions
### Proposed change

INatsClient Services API extensions

### Use case

INatsClient Services API extensions

Note: Expose INatsConnection as a property

### PR DESCRIPTION
**Pull request by mtmk**
Refactor to use interfaces for enhanced flexibility and testability across various components like NatsJSContext and NatsConnection. Added new extension methods to easily create contexts for Object Store, Key-Value Store, and Services on NATS client and connection instances.

```csharp
await using var client = new NatsClient();

// Use KeyValueStore extension method by referencing NATS.Client.KeyValueStore package
INatsKVContext kv = client.CreateKeyValueStoreContext();

// Use ObjectStore extension method by referencing NATS.Client.ObjectStore package
INatsObjContext obj = client.CreateObjectStoreContext();

// Use Services extension method by referencing NATS.Client.Services package
INatsSvcContext svc = client.CreateServicesContext();
```

* fixes #613 
* fixes #614 
* fixes #615

### CODE REVIEW

**Comment by rickdotnet** [comment]
:raised_hands: looking forward to this!

File: src/NATS.Client.KeyValueStore/INatsKVContext.cs [hunk 1]
```diff
@@ -4,6 +4,11 @@ namespace NATS.Client.KeyValueStore;
 
 public interface INatsKVContext
 {
+    /// <summary>
+    /// Provides access to the JetStream context associated with the Key-Value Store operations.
+    /// </summary>
+    INatsJSContext Context { get; }
```
Comment (rickdotnet) [inline src/NATS.Client.KeyValueStore/INatsKVContext.cs:10]: This makes sense to me, but wondering what you think about folks using code like

```cs
var context = client.CreateKeyValueStoreContext();
...
var jsContext = context.Context;
```

I don't mind it, but the thought came to mind and figured I'd mention it.

File: src/NATS.Client.KeyValueStore/NatsKVContext.cs [hunk 2]
```diff
@@ -23,13 +23,14 @@ public class NatsKVContext : INatsKVContext
...
-    private readonly NatsJSContext _context;
-
     /// <summary>
     /// Create a new Key Value Store context
     /// </summary>
     /// <param name="context">JetStream context</param>
-    public NatsKVContext(NatsJSContext context) => _context = context;
+    public NatsKVContext(INatsJSContext context) => Context = context;
+
+    /// <inheritdoc />
```
Comment (rickdotnet) [inline src/NATS.Client.KeyValueStore/NatsKVContext.cs:32]: My OCD thanks you for this. Prob don't need to, but could do the rest while you're here. ;)  Might make sense as a separate maintenance PR.

File: src/NATS.Client.KeyValueStore/NatsClientExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+public static class NatsClientExtensions
+{
+    /// <summary>
+    /// Creates a NATS Key-Value Store context using the specified NATS client.
+    /// </summary>
+    /// <param name="client">The NATS client instance.</param>
+    /// <returns>An instance of <see cref="INatsKVContext"/> which can be used to interact with the Key-Value Store.</returns>
+    public static INatsKVContext CreateKeyValueStoreContext(this INatsClient client)
+        => CreateKeyValueStoreContext(client.Connection);
```
Comment (rickdotnet) [inline src/NATS.Client.KeyValueStore/NatsClientExtensions.cs:14]: Does it make sense to chain these to the JS extensions?

```cs
public static INatsKVContext CreateKeyValueStoreContext(this INatsClient client)
    => CreateKeyValueStoreContext(client.CreateJetStreamContext());

public static INatsKVContext CreateKeyValueStoreContext(this INatsConnection connection)
    => CreateKeyValueStoreContext(connection.CreateJetStreamContext());
```

**Review by rickdotnet — APPROVED** [review]
LGTM. All comments are thought provoking.

File: src/NATS.Client.KeyValueStore/INatsKVContext.cs (hunk 1 above)
Comment (mtmk) [reply src/NATS.Client.KeyValueStore/INatsKVContext.cs:10]: What about `JetStreamContext` or just `JetStream`?
```csharp
var context = client.CreateKeyValueStoreContext();
...
var jsContext = context.JetStreamContext;
...
var jsContext = context.JetStream;
```

File: src/NATS.Client.KeyValueStore/NatsKVContext.cs (hunk 2 above)
Comment (mtmk) [reply src/NATS.Client.KeyValueStore/NatsKVContext.cs:32]: Do you mean using the new type constructors (or whatever they're called)?

File: src/NATS.Client.KeyValueStore/NatsClientExtensions.cs (hunk 3 above)
Comment (mtmk) [reply src/NATS.Client.KeyValueStore/NatsClientExtensions.cs]: makes sense 💯

File: src/NATS.Client.KeyValueStore/NatsKVContext.cs (hunk 2 above)
Comment (rickdotnet) [reply src/NATS.Client.KeyValueStore/NatsKVContext.cs:32]: ```/// <inheritdoc />```

File: src/NATS.Client.KeyValueStore/INatsKVContext.cs (hunk 1 above)
Comment (rickdotnet) [reply src/NATS.Client.KeyValueStore/INatsKVContext.cs:10]: I like `JetStreamContext`, if there are no conventions in play. The `INatsObjStore` version should probably share the name, if that changes things.

File: src/NATS.Client.KeyValueStore/NatsKVContext.cs (hunk 2 above)
Comment (mtmk) [reply src/NATS.Client.KeyValueStore/NatsKVContext.cs:32]: fixed

File: src/NATS.Client.KeyValueStore/INatsKVContext.cs (hunk 1 above)
Comment (mtmk) [reply src/NATS.Client.KeyValueStore/INatsKVContext.cs]: done

**Pull request merged**